media-api/app/lib/ImageResponse.scala

```diff
@@ -363,6 +364,22 @@ class ImageResponse(config: MediaApiConfig, s3Client: S3Client, usageQuota: Usag
   }
 }
 
+object ImageResponse {
+  val newlineNormalisingImageMetadataWriter: Writes[ImageMetadata] = (input: ImageMetadata) => {
+    Json.toJson(normaliseNewLines(input))
+  }
+
+  def normaliseNewLines(imageMetadata: ImageMetadata): ImageMetadata = imageMetadata.modifyAll(
+    _.description,
+    _.copyrightNotice,
+    _.copyright,
+    _.specialInstructions,
+    _.suppliersReference
+  ).using(_.map(ImageResponse.normaliseNewLines))
+
+  def normaliseNewLines(string: String): String = string.replaceAll("(\\r|\\n|\\r\\n)+", "\n")
```

Comment:
using `String.replaceAll()` will compile the regular expression **each time** you call it, as implementation is 
```java
public String replaceAll(String regex, String replacement) {
        return Pattern.compile(regex).matcher(this ).replaceAll(
             replacement);
}
```
One can use a pattern instead
```scala
private val pattern = """(\r|\n|\r\n)+""".r
def normaliseNewLines(string: String): String = pattern.replaceAllIn(string, "\n")
```

I think using `"""` ease readability as well

Reply:
I considered this as I wrote it but concluded it was premature optimisation and that the compiler or runtime would optimise this away given that it knows the parameter is a constant. I'll run some performance comparison tests tomorrow to see what the impact is given your concern.

Reply:
I am quite confident neither the compiler nor runtime will optimise this, as non using a pattern is a known performance bottleneck, for instance see [3.5 precompiled pattern performance benchmark](https://www.baeldung.com/java-string-performance). You will find as well the same advice (including using triple quotes for readability) [on the Scala documentation](https://www.scala-lang.org/api/2.12.8/scala/util/matching/Regex.html)
```
An instance of Regex represents a compiled regular expression pattern.
Since compilation is expensive, frequently used Regexes should be constructed once,
outside of loops and perhaps in a companion object.
```

Change as well does not impact readability, so could be better qualified as a best practice rather than a premature optimisation.

Reply:
I ran some numbers. It shaves around 15% off the clock time but will also be reducing object allocation (although that's likely to be insignificant given the context).

Reply:
I've continued my testing and have also written a version in fastparse. It's about 4 times faster. But also hard to read so I'm not going to use it.